mirgecom/boundary.py

```diff
@@ -1416,8 +1529,112 @@ def adiabatic_slip_grad_av(self, dcoll, dd_bdry, grad_av_minus, **kwargs):
                                   np.dot(grad_av_minus.momentum, nhat))
         s_mom_flux = grad_av_minus.momentum - 2*s_mom_normcomp
 
-        # flip components to set a neumann condition
+        # flip components to set a Neumann condition
         return make_conserved(dim, mass=-grad_av_minus.mass,
                               energy=-grad_av_minus.energy,
                               momentum=-s_mom_flux,
                               species_mass=-grad_av_minus.species_mass)
+
+
+class LinearizedOutflowBoundary(PrescribedFluidBoundary):
+    r"""Characteristics outflow BCs for linearized Euler equations.
+
+    Implement non-reflecting outflow based on characteristic variables for
+    the Euler equations assuming small perturbations based on [Giles_1988]_.
+    The equations assume an uniform, steady flow and linerize the Euler eqs.
+    in this reference state, yielding a linear equation in the form
+
+    .. math::
+        \frac{\partial U}{\partial t} + A \frac{\partial U}{\partial x} +
+        B \frac{\partial U}{\partial y} = 0
+
+    where where U is the vector of perturbation (primitive) variables and
+    the coefficient matrices A and B are constant matrices based on the
+    uniform, steady variables.
+
+    Using the linear hyperbolic system theory, this equation can be further
+    simplified by ignoring the y-axis terms (tangent) such that wave propagation
+    occurs only along the x-axis direction (normal). Then, the eigendecomposition
+    results in a orthogonal system where the wave have characteristic directions
+    of propagations and enable the creation of non-reflecting outflow boundaries.
+
+    This can also be applied for Navier-Stokes equations in regions where
+    viscous effects are not dominant, such as the far-field.
+    """
+
+    def __init__(self, free_stream_state=None,
+                 free_stream_density=None,
+                 free_stream_velocity=None,
+                 free_stream_pressure=None,
+                 free_stream_temperature=None,
+                 free_stream_species_mass_fractions=None):
+        """Initialize the boundary condition object."""
+        self._ref_state = free_stream_state
+        self._mass = free_stream_density
+        self._velocity = free_stream_velocity
+        self._pressure = free_stream_pressure
+        self._temperature = free_stream_temperature
+        self._Y = free_stream_species_mass_fractions
```

Comment:
How's come this?   Why not s'th like:
```suggestion
if free_stream_state is not None:
   self._mass = free_stream_state.mass
   (...) etc
```
... and then eliminate the if/else blocks that check for existence of `self._ref_state`?